pkg/distributor/ha_tracker_test.go

```diff
@@ -897,3 +897,36 @@ func getSumOfHistogramSampleCount(families []*dto.MetricFamily, metricName strin
 
 	return sum
 }
+
+// Test that values are synced to KVStore on starting
+func TestStartingSync(t *testing.T) {
+	cluster := "c1"
+	replica := "r1"
+
+	codec := GetReplicaDescCodec()
+	kvStore, closer := consul.NewInMemoryClient(codec, log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	mock := kv.PrefixClient(kvStore, "prefix")
+	c, err := newHATracker(HATrackerConfig{
+		EnableHATracker:        true,
+		KVStore:                kv.Config{Mock: mock},
+		UpdateTimeout:          15 * time.Second,
+		UpdateTimeoutJitterMax: 0,
+		FailoverTimeout:        time.Millisecond * 2,
+	}, trackerLimits{maxClusters: 100}, nil, log.NewNopLogger())
+	require.NoError(t, err)
+	require.NoError(t, services.StartAndAwaitRunning(context.Background(), c))
+	defer services.StopAndAwaitTerminated(context.Background(), c) //nolint:errcheck
+
+	now := time.Now()
+
+	err = c.starting(context.Background())
+	assert.NoError(t, err)
+
+	err = c.checkReplica(context.Background(), "user", cluster, replica, now)
+	assert.NoError(t, err)
```

Comment:
Isn't this also the behaviour (i.e. no error) when the initial sync doesn't happen? IIRC the HA tracker will accept the replica and return no error when it cannot find it in the local cache. 

It would be more interesting to test that a write for a secondary replica is rejected with a `replicasNotMatchError`. One way to to set up the test is to 
* start an HA tracker
* check one replica
* shut down the HA tracker, but keep the KV store running
* start another HA tracker
* check a different replica for the same cluster & expect that it's rejected with a `replicasNotMatchError`

This proves that a new HA tracker that hasn't received any writes yet has an up-to-date view that it synced from the KV store upon startup.